### DISCUSSION TITLE
Support importing accounts from the URL exported by Google Authenticator export QR code

### DISCUSSION THREAD

**Discussion by 05storm26** [dani-garcia/vaultwarden#3998] [Ideas]
Support importing accounts from the URL exported by Google Authenticator export functionality.
The export generates a QR code that contains within an URL like this:  
```
otpauth-migration://offline?data=urlencoded-base64-encoded-protobuff-binary-data
```

The protobuff protocol specifing the content is this:

```protobuf
syntax = "proto3";

// Copied from: https://github.com/beemdevelopment/Aegis/blob/master/app/src/main/proto/google_auth.proto

message MigrationPayload {
  enum Algorithm {
    ALGO_INVALID = 0;
    ALGO_SHA1 = 1;
  }

  enum OtpType {
    OTP_INVALID = 0;
    OTP_HOTP = 1;
    OTP_TOTP = 2;
  }

  message OtpParameters {
    bytes secret = 1;
    string name = 2;
    string issuer = 3;
    Algorithm algorithm = 4;
    int32 digits = 5;
    OtpType type = 6;
    int64 counter = 7;
  }

  repeated OtpParameters otp_parameters = 1;
  int32 version = 2;
  int32 batch_size = 3;
  int32 batch_index = 4;
  int32 batch_id = 5;
}

````

It would be great if there were an password import type functionality that could import totp accounts from this qr code / migration URL into vaultwarden.

**Comment by stefan0xC** [comment]
That is not really something that we can implement. It would have to be implemented in the [clients](https://github.com/bitwarden/clients) themselves.

There already is [a feature request in the Bitwarden community forum](https://community.bitwarden.com/t/import-2fa-accounts-from-google-authenticator-into-bitwarden/45588) which also links to [a thread that discusses how to import TOTP from Google Authenticator](https://community.bitwarden.com/t/import-totp-from-google-authenticator/44996/4).
